jobs/e2e-test-cypress.groovy

```diff
@@ -0,0 +1,138 @@
+// Vars we use, under jenkins-jobs/vars/.  This is just for documentation.
+//import vars.exec
+//import vars.notify
+//import vars.withTimeout
+//import vars.withSecrets
+
+public class TestFailed extends Exception {}
+
+new Setup(steps
+
+).addStringParam(
+   "GIT_REVISION",
+   """The name of a webapp branch to use when building.""",
+   "master"
+
+).addStringParam(
+   "URL",
+   "The url-base to run these tests against.",
+   "https://www.khanacademy.org/"
+
+).addStringParam(
+   "SLACK_CHANNEL",
+   "The slack channel to which to send failure alerts.",
+   "#cypress-testing"
+
+).addStringParam(
+   "DEPLOYER_USERNAME",
+   """Who asked to run this job, used to ping on slack.
+Typically not set manually, but rather by other jobs that call this one.""",
+   "username"
+
+).addStringParam(
+   "REVISION_DESCRIPTION",
+   """Set by the buildmaster to give a more human-readable description
+of the GIT_REVISION, especially if it is a commit rather than a branch.
+Defaults to GIT_REVISION.""",
+   ""
+
+).addStringParam(
+   "NUM_WORKER_MACHINES",
+   """How many worker machines to use. Max available is 20.""",
+   "20"
+
+).addStringParam(
+   "TEST_RETRIES",
+   """How many retry attempts to use. By default is 1.""",
+   "1"
+
+).addChoiceParam(
+   "TEST_TYPE",
+   """\
+<ul>
+  <li> <b>all</b>: run all tests</li>
+  <li> <b>deploy</b>: run only those tests that are important to run at
+        deploy-time (as identified by the `@`
+        decorator)</li>
+  <li> <b>nightly</b>: run only night list of tests.</li>
+</ul>
+""",
+   ["all", "deploy", "nightly"]
+
+
+).apply()
+
+def _setupWebapp() {
+   GIT_SHA1 = kaGit.resolveCommitish("git@github.com:Khan/webapp",
+                                     params.GIT_REVISION);
+   
+   kaGit.safeSyncToOrigin("git@github.com:Khan/webapp", GIT_SHA1);
+
+   dir("webapp") {
+      sh("make clean_pyc");
+
+   }
+}
+
+def runLamdaTest() {
+    _setupWebapp();
+
+    // We need login creds for LambdaTest Cli
+    def lt_username = sh(script: """\
+        keeper --config ${exec.shellEscape("${HOME}/.keeper-config.json")} \
+        get qvYpo_KnpCiLBN69fYpEYA --format json | jq -r .login\
+        """, returnStdout:true).trim();
+    def lt_access_key = sh(script: """\
+        keeper --config ${exec.shellEscape("${HOME}/.keeper-config.json")} \
+        get qvYpo_KnpCiLBN69fYpEYA --format json | jq -r .password\
+        """, returnStdout:true).trim();
+    
+    def runLambdaTestArgs = ["--cy=\"--config baseUrl=${params.URL}\", retries=${params.TEST_RETRIES}",
+                             "--bn",
+                             "e2e-test # (${params.URL}: ${params.REVISION_DESCRIPTION}) @${params.DEPLOYER_USERNAME}",
+                             "-p",
+                             "${params.NUM_WORKER_MACHINES}"
+    ];
+    
+    try {
+        dir('webapp/services/static') {
+            
+            sh("git checkout feature/cypress");
+            sh("yarn install")
```

Comment:
It's not good practice to pollute the webapp workspace like this.  In particular, I think this will fail if you happen to run on a worker machine that already has feature/cypress set up.  You should be using `safeSyncToOrigin` instead, like with webapp.  And do that in workspace-root, so cypress is a sibling dir to webapp instead of embedded inside it.

(And probably you should take a CYPRESS_GIT_REVISION so people can run a non-master version of cypress if they want.)

I don't know how `yarn lambdatest` will work in that situation -- does it assume that feature/cypress is checked out in a particular location?  If so you may need to mess with that so it can find cypress out-of-tree.